pyuvdata/calfits.py

```diff
@@ -0,0 +1,194 @@
+
+        for p in self.extra():
+            ep = getattr(self, p)
+            if ep.form is 'str':
+                prihdr['{0}'.format(p.upper().replace('_', '')[:8])] = ep.value
+            else:
+                continue
+
+        # if self.pipeline: prihdr['CALPIPE'] = self.pipeline
+        # if self.observer: prihdr['OBSERVER'] = self.observer
+        # if self.git_origin: prihdr['ORIGIN'] = self.git_origin
+        # if self.git_hash: prihdr['HASH'] = self.git_hash
+
+        if self.cal_type == 'gain':
+            # Set header variable for gain.
+            prihdr['CTYPE4'] = ('FREQS', 'Frequency.')
+            prihdr['CUNIT4'] = ('Hz', 'Units of frequecy.')
+            prihdr['CRVAL4'] = self.freq_array[0][0]
+            prihdr['CDELT4'] = self.channel_width
+
+            # set the last axis for number of arrays.
+            prihdr['CTYPE1'] = ('Narrays', 'Number of image arrays.')
+            prihdr['CUNIT1'] = ('Integer', 'Number of image arrays. Increment.')
+            prihdr['CRVAL1'] = (3, 'Number of image arryays.')
+            prihdr['CDELT1'] = 1
+
+            pridata = np.concatenate([self.gain_array.real[:, :, :, :, np.newaxis],
+                                      self.gain_array.imag[:, :, :, :, np.newaxis],
+                                      # self.flag_array[:, :, :, :, np.newaxis],
+                                      self.quality_array[:, :, :, :, np.newaxis]],
+                                     axis=-1)
+
+        if self.cal_type == 'delay':
+            # Set header variable for gain.
+            prihdr['CTYPE4'] = ('FREQS', 'Valid frequencies to apply delay.')
+            prihdr['CUNIT4'] = ('Hz', 'Units of frequecy.')
+            prihdr['CRVAL4'] = self.freq_array[0][0]
+            prihdr['CDELT4'] = self.channel_width
+            # set the last axis for number of arrays.
+            prihdr['CTYPE1'] = ('Narrays', 'Number of image arrays.')
+            prihdr['CUNIT1'] = ('Integer', 'Number of image arrays. Value.')
+            prihdr['CRVAL1'] = (2, 'Number of image arrays.')
+            prihdr['CDELT1'] = 1
+
+            pridata = np.concatenate([self.delay_array[:, :, :, :, np.newaxis],
+                                      # self.flag_array[:, :, :, :, np.newaxis],
+                                      self.quality_array[:, :, :, :, np.newaxis]],
+                                     axis=-1)
+
+        if self.cal_type == 'unknown':
+            raise ValueError("unknown calibration type. Do not know how to"
+                             "store parameters")
+
+        # header ctypes for NAXIS
+        prihdr['CTYPE5'] = ('ANTS', 'Antenna numbering.')
+        prihdr['CUNIT5'] = 'Integer'
+        prihdr['CRVAL5'] = 0
+        prihdr['CDELT5'] = -1
+
+        prihdr['CTYPE3'] = ('TIME', 'Time axis.')
+        prihdr['CUNIT3'] = ('JD', 'Time in julian date format')
+        prihdr['CRVAL3'] = self.time_array[0]
+        prihdr['CDELT3'] = self.integration_time
```

Comment:
I think that could work, but I'd lean toward something more explicit, like a freq_range and time_range with two numbers with units.

In addition to clarity, one reason that I prefer this approach is that it's not clear to me that times in uvdata objects all have to be equally separated (and indeed we have discussed doing cat's eye averaging for HERA data). Of course, if that did happen we'd have to make some other changes to cal objects and files, so this might not be something to worry about now.